subprojects/core/src/main/java/org/gradle/api/internal/project/MutableStateAccessAwareProject.java

```diff
@@ -138,6 +141,16 @@ protected MutableStateAccessAwareProject(ProjectInternal delegate) {
     @SuppressWarnings("unused") // used by Groovy dynamic dispatch
     protected abstract Object methodMissing(String name, Object args);
 
+    protected void propertyMissing(String name, Object args) {
```

Comment:
💅  We should add the comment that this is related to Groovy dynamic dispatch

Why is this method implemented here, while the getter-related method stays abstract is expected to be implemented by child classes?

Reply:
Because of setProperty doing the same for both implementations, as well as `hasProperty`.
But `getProperty` has different implementations